src/infrastructure/config/app_config.py

```diff
@@ -1,15 +1,46 @@
 from os import getenv
-from pydantic import BaseModel
+from functools import lru_cache
+from dotenv import load_dotenv
 from async_fastapi_jwt_auth import AuthJWT
+from pydantic import BaseModel, BaseSettings
 
-DATABASE_URL = getenv('DATABASE_URL')
-SECRET_KEY = '3f23d9abf331784b834b62aa347cfaf7cd1970f0c072d9d4e8595e41f5e708c5'
-ALGORITHM = 'HS256'
-STATIC_DIR = getenv('STATIC_DIR', default='../static')
-ALLOWED_ORIGINS = ['*']
-ALLOWED_METHODS = ['*']
-ALLOWED_HEADERS = ['*']
-ALLOW_CREDENTIALS = False
+load_dotenv()
+
+if getenv('LOCAL') == "1":
+    class FastAPISettings(BaseSettings):
```

Comment:
klasę i metodę wyciągnąłbym z instrukcji warunkowej.
Warunki byłyby takie raczej: if (getenv('ENV') == 'LOCAL'), elif (getenv('ENV') == 'DOCKER') i else
I nie wyciągałbym tego do zmiennych globalnych tylko korzystał już z FastApiSettings, chociaż bardziej można by to było nazwać po prostu ApplicationSettings czy coś takiego
Tutaj masz link, z dwoma sposobami wciągania plików .env. Druga metoda nam by pasowała, ze względu na if else i możliwość wskazania skąd ma wciągnąć. To sprawia, że load_dotenv jest useless w 7 linijce 
https://pydantic-docs.helpmanual.io/usage/settings/#dotenv-env-support

Reply:
Czyli w skrócie, jak ENV == local to pobierasz z np. `.local.env`
Jak ENV == docker to pobierasz z np. `.docker.env`
A w innym przypadku tworzysz klasę settingsów z danymi z getenv